hazelcast/src/main/java/com/hazelcast/map/impl/MapMigrationAwareService.java

```diff
@@ -174,16 +174,20 @@ public void commitMigration(PartitionMigrationEvent event) {
             depopulateIndexes(event, "commitMigration");
         }
 
+        PartitionContainer partitionContainer
+                = mapServiceContext.getPartitionContainer(event.getPartitionId());
         if (SOURCE == event.getMigrationEndpoint()) {
             // Do not change order of below methods
             removeWbqCountersHavingLesserBackupCountThan(event.getPartitionId(),
                     event.getNewReplicaIndex());
             removeRecordStoresHavingLesserBackupCountThan(event.getPartitionId(),
                     event.getNewReplicaIndex());
+
+            if (event.getNewReplicaIndex() == -1) {
```

Comment:
also we can add `|| event.getNewReplicaIndex() > backupCount` since partitionContainer can have different maps.

Reply:
Thanks for reminding. Added in the cleanup method since we iterate over the maps there.